### ISSUE TITLE
Regarding the upsampling layer

### ISSUE THREAD

**Issue by netw0rkf10w** [aurora95/Keras-FCN#83]
Hi,

Thanks a lot for the implementation!

Regarding upsampling, you used a custom BilinearUpSampling2D layer, which I guess cannot be learned (sorry if I'm mistaken).

I believe a better solution is to use the built-in Conv2DTranspose Keras layer for upconvolution (c.f. section 3.3 of the [FCN paper](https://people.eecs.berkeley.edu/~jonlong/long_shelhamer_fcn.pdf)).

Best regards.

**Comment by netw0rkf10w** [comment]
I have checked the code again and there seems to have a more serious problem: there is no skip connections at all between coarse and fine output layers :(

**Comment by NickChang97** [comment]
Yes I found it also. There is no skip connections, a little pity. But the code's structure is nice
